cmd/list.go

```diff
@@ -0,0 +1,61 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	analytics "github.com/segmentio/analytics-go"
+	"github.com/segmentio/aws-okta/lib"
+	"github.com/spf13/cobra"
+)
+
+// listCmd represents the list command
+var listCmd = &cobra.Command{
+	Use:   "list",
+	Short: "list will show you the profiles currently configured",
+	RunE:  listRun,
+}
+
+func init() {
+	RootCmd.AddCommand(listCmd)
+}
+
+func listRun(cmd *cobra.Command, args []string) error {
+	config, err := lib.NewConfigFromEnv()
+	if err != nil {
+		return err
+	}
+
+	profiles, err := config.Parse()
+	if err != nil {
+		return err
+	}
+
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 8, 2, '\t', 0)
+	fmt.Fprintln(w, "profile\tarn\tsource_role\t")
+	fmt.Fprintln(w, "---\t---\t---\t")
+	for profile, v := range profiles {
+		if role, exist := v["role_arn"]; exist {
+			if src, exist := v["source_profile"]; exist {
```

Comment:
In the case where there is no `source_profile`, no line gets printed. This is the case in my config, so I get no results from the whole command :( I wonder if you could use `lib/config.go`'s `sourceProfile` function?